### PR TITLE
pkg/system, pkg/mount: assorted improvements

### PR DESCRIPTION
**Pull request by kolyshkin**
* improved error reporting for mount.Mount()/Unmount() and system.[L]Stat()
* dropped Solaris and FreeBSD support for pkg/mount
* optimizations for `system.EnsureRemoveAll()` and `mount.RecursiveUnmount()` (same as https://github.com/moby/moby/pull/40637)

### CODE REVIEW

File: pkg/system/rm.go [hunk 1]
```diff
@@ -63,12 +63,8 @@ func EnsureRemoveAll(dir string) error {
...
 		}
 
-		if mounted, _ := mount.Mounted(pe.Path); mounted {
-			if e := mount.Unmount(pe.Path); e != nil {
-				if mounted, _ := mount.Mounted(pe.Path); mounted {
-					return errors.Wrapf(e, "error while removing %s", dir)
-				}
-			}
+		if e := mount.Unmount(pe.Path); e != nil {
+			return errors.Wrapf(e, "error while removing %s", dir)
```
Comment (rhatdan) [inline pkg/system/rm.go]: In the existing code, the Unmount command would not be called, and we would not get an error if the code was not Mounted,  Why not check if the error is EINVAL, and then check if it is Mounted, and not return an error.
Then you are fast in the default path, and continue the functionality of not returning EINVAL errors.

File: pkg/mount/mount.go [hunk 2]
```diff
@@ -116,9 +116,7 @@ func RecursiveUnmount(target string) error {
 			continue
 		}
 		if err := Unmount(m.Mountpoint); err != nil && i == len(mounts)-1 {
-			if mounted, err := Mounted(m.Mountpoint); err != nil || mounted {
-				return err
-			}
+			return err
```
Comment (rhatdan) [inline pkg/mount/mount.go]: We do not want to bubble errors when some other tool unmounted the file system.

File: pkg/system/rm.go (hunk 1 above)
Comment (kolyshkin) [reply pkg/system/rm.go]: `mount.Unmount()` already ignores `EINVAL`

File: pkg/mount/mount.go (hunk 2 above)
Comment (kolyshkin) [reply pkg/mount/mount.go]: same here, `Mount()` ignores EINVAL

**Comment by kolyshkin** [comment]
Do we want to support darwin here? Currently, CI is failing on darwin as I removed some support for "unsupported" platforms.

**Comment by rhatdan** [comment]
Yes, when Container/Storage gets vendored into podman, it currently gets installed for MAC/WIndows platforms. Eventually this should be cleaned up, since we don't actually support C/S on a MAC, but for now podman-remote requires this to compile correctly.

**Comment by rhatdan** [comment]
LGTM
@giuseppe @vrothberg @nalind @TomSweeneyRedHat PTAL

**Pull request merged**